infra/gcp/istio-prow-private/ar.tf

```
@@ -0,0 +1,25 @@
+resource "google_artifact_registry_repository" "main" {
+  location      = "us"
+  repository_id = "istio-prow-private"
+  description   = "registry to host private Istio release artifacts"
+  format        = "DOCKER"
+
+  docker_config {
+    immutable_tags = true
+  }
```

all repos and buckets should have prevent destroy. this will prevent a terraform apply or delete from removing the resource
```suggestion
  }
    lifecycle {
    prevent_destroy = true
  }
```